### ISSUE TITLE
`roc dev` crashes with unhandled parse error

### ISSUE THREAD

**Issue by lukewilliamboswell** [roc-lang/roc#4777]
```sh
% roc dev test.roc
thread 'main' panicked at 'Error in alias analysis: error in module ModName("UserApp"), function definition FuncName("\x01\x00\x00\x00\x10\x00\x00\x00##7G\xc1/\x9e\x8e"), definition of value binding ValueId(6): expected type '((),)', found type '((heap_cell, bag<()>),)'', crates/compiler/gen_llvm/src/llvm/build.rs:4304:19
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

```elixir
app "aoc-2022"
    packages { pf: "https://github.com/roc-lang/basic-cli/releases/download/0.1.2/3bKbbmgtIfOyC6FviJ9o8F8xqKutmXgjCJx3bMfVTSo.tar.br" }
    imports [
        pf.Stdout,
        pf.Task.{ Task },
    ]
    provides [ main ] to pf

main : Task {} []
main =
    Stdout.line (part1 { data : [] })

part1 = \state ->

    dbg state
    
    sensorData =
        state 
        |> .data

    dbg sensorData

    ""
```

**Comment by lukewilliamboswell** [comment]
Also may be related?? Can log in separate issue if required.

```sh
% roc dev test.roc
thread 'main' panicked at 'not yet implemented: unhandled parse error: Dbg(Continuation(Start(@347), @341), @332)', crates/reporting/src/error/parse.rs:576:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

```elixir
app "aoc-2022"
    packages { pf: "https://github.com/roc-lang/basic-cli/releases/download/0.1.2/3bKbbmgtIfOyC6FviJ9o8F8xqKutmXgjCJx3bMfVTSo.tar.br" }
    imports [
        pf.Stdout,
        pf.Task.{ Task },
    ]
    provides [ main ] to pf

main : Task {} []
main =
    Stdout.line (part1 { data : [] })

part1 = \state -> 
    dbg state
```

**Comment by ayazhafiz** [comment]
The second is an error in the parser - the first should be solved by #4768 !

**Comment by lukewilliamboswell** [comment]
Confirmed the first is resolved by #4768

**Comment by lukewilliamboswell** [comment]
Renamed to focus this issue on the second, rather than re-open a new issue.
